app/scripts/lib/app-start.js

```diff
@@ -232,7 +220,17 @@ function (
     initializeMetrics: function () {
       var relier = this._relier;
       var screenInfo = new ScreenInfo(this._window);
-      this._metrics = createMetrics(this._config.metricsSampleRate, {
+      // Randomly choose whether to send metrics based on sample rate,
+      // but use the relier's state management to remember this choice
+      // throughout the entire flow.
+      var sendMetrics;
+      if (relier.has('metrics')) {
+        sendMetrics = relier.get('metrics');
+      } else {
+        sendMetrics = Math.random() <= this._config.metricsSampleRate;
+        relier.set('metrics', sendMetrics);
```

Comment:
I'll need this for my AB testing stuff, I'm thinking the sample rate is eligibility for the `eligibilityFunction` in able.